youtube_dl/extractor/americastestkitchen.py

```diff
@@ -0,0 +1,53 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+
+
+class AmericasTestKitchenIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?americastestkitchen\.com/episode/(?P<id>\d{3}(-\w+)+)'
+    _TESTS = [{
+        'url':
+        'https://www.americastestkitchen.com/episode/548-summer-dinner-party',
+        'md5': 'b861c3e365ac38ad319cfd509c30577f',
+        'info_dict': {
+            'id': '1_5g5zua6e',
+            'title': 'atk_s17_e24.mp4',
+            'ext': 'mp4',
+            'description': 'S17',
+            'timestamp': 1497285541,
+            'upload_date': '20170612',
+            'uploader_id': "roger.metcalf@americastestkitchen.com",
+        },
+    }, {
+        'url':
+        'https://www.americastestkitchen.com/episode/546-a-spanish-affair',
+        'only_matching':
+        True,
+    }]
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('id')
+        # display_id = mobj.group('display_id')
+        webpage = self._download_webpage(url, video_id)
+
+        partner_id = self._html_search_regex(
+            r'(?:partner_id)/(?P<partner_id>\d+)',
+            webpage,
+            'partner_id',
+            default=None)
+        kaltura_id = self._search_regex(
+            r'([\"\'])external_id\1\s*[:=]\1(?P<id>[0-9a-z_]+)\1',
+            webpage,
+            'video_id',
+            group='id')
```

Comment:
This matches multiple videos.

Reply:
I'm not seeing what you are seeing with this. When I debug & look at 

https://github.com/rg3/youtube-dl/blob/2cfa7cbdd09353c8f8b135ba1c0b3a325bb03a16/youtube_dl/extractor/common.py#L788-L791

I only see partnerId hitting line 791, where groups is populated, however it's length is 1. Are you still counting this as multiple matches? 

Secondly when I run my tests I get something like 
```
[AmericasTestKitchen] 548: Downloading webpage
[Kaltura] 1_5g5zua6e: Downloading video info JSON
[Kaltura] 1_5g5zua6e: Downloading m3u8 information
[info] Writing video description metadata as JSON to: test_AmericasTestKitchen_1_5g5zua6e.info.json
[debug] Invoking downloader on u'http://cdnapi.kaltura.com/p/1445801/sp/144580100/playManifest/entryId/1_5g5zua6e/format/url/protocol/http/flavorId/1_g8znrmp2'
[download] Destination: test_AmericasTestKitchen_1_5g5zua6e.mp4
[download] 100% of 10.00KiB in 00:00
```
which to me looks like it is only matching a single video. 

Am I wrong in my understanding here?

Reply:
`r'([\"\'])external_id\1\s*[:=]\1(?P<id>[0-9a-z_]+)\1'` matches **several times** on the webpage meaning that **wrong video will be downloaded** if actual video happens **not to be the first matched**.

Reply:
Ok so I changed my method of getting the required data. I'm now pulling JSON from a script tag parsing that & using it's values. This restricts the searching for this external_id to one specific field. It also allows me to pull more data out that is useful.